### PR TITLE
Added filterWhere function to FindWrapper

### LINKED ISSUE #54: filterWhere function for FindWrapper
It would be nice to have a filterWhere function in FindWrapper, that mimic's enzyme's functionality.

### PR DESCRIPTION
**Pull request by DzikowskiW**
`filterWhere` function was added mimicking enzyme's functionality
Resolves #54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,9 @@ Returns a new `FindWrapper` with a subset of the previously selected elements gi
 
 Uses the same possible selectors as [`RenderContext#find(selector)`](#rendercontextfindselector).
 
+### `FindWrapper#filterWhere(predicate)`
+Returns a new `FindWrapper` with a subset of the previously selected elements which, when passed into the provided predicate function, return true.
```
Comment (gnarf) [inline README.md]: A small usage example here would be useful.

**Comment by gnarf** [comment]
Also 😍 thanks for all the submissions!

**Comment by mzgoddard** [comment]
I concur with https://github.com/mzgoddard/preact-render-spy/pull/55#pullrequestreview-71558922 @gnarf. We can overload `find` and `filter` with the function style matching semantics.

**Comment by DzikowskiW** [comment]
I'm sorry for the delay, I had a hectic end of the month, I will have the changes ready by the end of the week. 

For this pull request I still need to update` README` and `index.d.ts`

**Pull request closed without merging**